rides-and-fares/src/main/java/org/apache/flink/training/exercises/ridesandfares/RidesAndFaresExercise.java

```diff
@@ -99,19 +102,43 @@ public static void main(String[] args) throws Exception {
     public static class EnrichmentFunction
             extends RichCoFlatMapFunction<TaxiRide, TaxiFare, RideAndFare> {
 
+        private ValueState<TaxiRide> taxiRideValueState;
+        private ValueState<TaxiFare> taxiFareValueState;
```

Comment:
it should be `transient`

Reply:
There was a discussion about transient states. But because this is a static class, there is no need.

Reply:
I missed that part. Where did you find this discussion?
your class `extends RichCoFlatMapFunction`which is Serializable
Flink use Serialization. So to avoid have issues you must mark ANY flink states as `transient`
because it has it's own Lifecycle see `open` method.

Reply:
It was in the module1 video. I need to look up it.
One thing that makes me wonder is, why the solutions do not have the fields marked as transient?

Reply:
But indeed, after testing it, I do get a `NotSerializableException` if the fields are not marked as transient.

Reply:
ok, rule number 1 in Flink everything must be Serializable.
 And if it's Pojo field and if it's Flink State (org.apache.flink.api.common.state.State) it has to be marked as a `transient` otherwise you will get in trouble.

Reply:
Please make it `transient` and always exclude any Flink state from Serialization/Deserialization process